files/image_config/warmboot-finalizer/finalize-warmboot.sh

```diff
@@ -18,6 +28,20 @@ function debug()
 }
 
 
+function set_component_list()
+{
+    CP_LIST=${!RECONCILE_COMPONENTS[@]}
+    COMPONENT_LIST=""
+    for cp in ${CP_LIST}; do
+        service=${RECONCILE_COMPONENTS[${cp}]}
+        status=$(show feature status | grep "^${service}" | awk '{ print $2 }')
```

Comment:
This logic depends on the raw output format of `show feature status`. If anything changes in this show api, `awk` might return some other column value. Can this rather be obtained directly from the DB?
```
$ redis-cli -n 4 HMGET "FEATURE|nat" state
1) "disabled"

$ redis-cli -n 4 HMGET "FEATURE|bgp" state
1) "enabled"

...
```

Reply:
good idea! changed. Thanks!